android-test/src/androidTest/java/okhttp/android/test/OkHttpTest.kt

```diff
@@ -91,7 +103,20 @@ class OkHttpTest {
         assertEquals(TlsVersion.TLS_1_2, response.handshake?.tlsVersion)
       }
       assertEquals(200, response.code)
-      assertEquals("com.android.org.conscrypt.Java8FileDescriptorSocket", socketClass)
+      assertTrue(socketClass?.startsWith("com.android.org.conscrypt.") == true)
+    }
+  }
+
+  @Test
+  fun testHttpRequestNotBlockedOnLegacyAndroid() {
+    Assume.assumeTrue(Build.VERSION.SDK_INT < 23)
+
+    val request = Request.Builder().url("http://api.twitter.com/robots.txt").build()
```

Comment:
nit: twitter is a weird choice. Maybe `squareup.com` instead? Our CEO is much better!